### DISCUSSION TITLE
Why LinqToDB stores Guids in SQLite as binary (both with System.Data.SQLite and Microsoft.Data.Sqlite provider)?

### DISCUSSION THREAD

**Discussion by alexey-leonovich** [linq2db/linq2db#4500] [Q&A] [answered]
Hello.
I've noticed that by default LinqToDB stores in SQLite as binary (both with System.Data.SQLite and Microsoft.Data.Sqlite provider).
EF Core switched from BLOB to TEXT some time ago but motivation is not clear enough for me. @MaceWindu, you [participated](https://github.com/dotnet/efcore/issues/15078#issuecomment-544833435) in that discussion - maybe you can clarify that? Is the only issue with Guid as BLOB was writing Guid to database via C# code and reading it from same database via Java? What are drawbacks of storing Guid as TEXT that make you not to follow EF Core command decision? Thank you.

**Comment by MaceWindu** [comment, marked as answer]
For historical reasons and backward compatibility. MS was provider introduced long after `System.Data.Sqlite` provider we supported initially. Plus it also used binary mapping for first versions
